src/box/lua/config/applier/credentials.lua

```diff
@@ -809,6 +836,56 @@ end
 -- of main fiber in case of 'BLOCKING_FULL_SYNC'.
 local wait_sync
 
+local function sync_object(obj_to_sync)
+    if obj_to_sync.type == 'BLOCKING_FULL_SYNC' or
+        obj_to_sync.type == 'BACKGROUND_FULL_SYNC' then
+
+        if box.info.ro then
+            log.verbose('credentials: the database is in the read-only ' ..
+                'mode. Waiting for the read-write mode to set up the ' ..
+                'credentials given in the configuration. If another ' ..
+                'instance in the replicaset will write these new ' ..
+                'credentials, this instance will receive them and will ' ..
+                'skip further actions.')
+
+            box.ctl.wait_rw()
+        end
+
+        if not schema_is_upgraded then
```

Comment:
Why do we need to always check it in a loop, on every new task? `schema_is_upgraded` can happen only after instance restart. It seems we should block `sync_credentials_worker` execution right after its start until the schema is upgraded.

Reply:
That'll be a lot cleaner. Thank you. Did it this way.